--- a/root/etc/config/luci
+++ b/root/etc/config/luci
@@ -1,6 +1,6 @@
 config core main
-	option lang auto
-	option mediaurlbase /luci-static/bootstrap
+	option lang 'zh_cn'
+	option mediaurlbase /luci-static/atmaterial
 	option resourcebase /luci-static/resources
 	
 config extern flash_keep
@@ -22,3 +22,7 @@ config internal ccache
 	option enable 1
 		
 config internal themes
+	option Bootstrap '/luci-static/bootstrap'
+	option Material '/luci-static/material'
+	option ADVancedtomato '/luci-static/advancedtomato'
+	option ATMaterial '/luci-static/atmaterial'
