--- a/luasrc/controller/admin/index.lua
+++ b/luasrc/controller/admin/index.lua
@@ -22,6 +22,10 @@ function index()
 	-- Empty services menu to be populated by addons
 	entry({"admin", "services"}, firstchild(), _("Services"), 40).index = true
 
+	entry({"admin","qos"},firstchild(),_("QOS"),41).index=true
+	entry({"admin","vpn"},firstchild(),_("VPN"),42).index=true
+	entry({"admin","control"},firstchild(),_("Control"),43).index=true
+
 	entry({"admin", "logout"}, call("action_logout"), _("Logout"), 90)
 end
 
