--- a/luasrc/controller/admin/system.lua
+++ b/luasrc/controller/admin/system.lua
@@ -18,6 +18,10 @@ function index()
 		entry({"admin", "system", "packages", "ipkg"}, form("admin_system/ipkg"))
 	end
 
+	if fs.access("/usr/sbin/shellinaboxd")then
+		entry({"admin","system","WebShell"},template("admin_system/webshell"),_("WebShell"),11)
+	end
+
 	entry({"admin", "system", "startup"}, form("admin_system/startup"), _("Startup"), 45)
 	entry({"admin", "system", "crontab"}, form("admin_system/crontab"), _("Scheduled Tasks"), 46)
 
