--- a/luasrc/model/cbi/admin_network/wifi.lua	2018-08-17 22:49:10.465155312 +0800
+++ b/luasrc/model/cbi/admin_network/wifi.lua	2018-08-18 01:38:46.034778760 +0800
@@ -254,6 +254,16 @@
 	s:taboption("advanced", Flag, "noscan", translate("Force 40MHz mode"),
 		translate("Always use 40MHz channels even if the secondary channel overlaps. Using this option does not comply with IEEE 802.11n-2009!")).optional = true
 
+
+	log_level = s:taboption("advanced", ListValue, "log_level", translate("Logging Level"))
+	log_level.default = "2"
+	log_level.optional = true
+	log_level:value("2", translate("Informational Message"))
+	log_level:value("0", translate("Verbose Debugging"))
+	log_level:value("1", translate("Debugging"))
+	log_level:value("3", translate("Notification"))
+	log_level:value("4", translate("Warning"))
+
 	beacon_int = s:taboption("advanced", Value, "beacon_int", translate("Beacon Interval"))
 	beacon_int.optional = true
 	beacon_int.placeholder = 100
