--- a/luasrc/model/cbi/aria2.lua	2018-05-29 20:35:31.570202402 +0800
+++ b/luasrc/model/cbi/aria2.lua	2018-10-07 14:08:12.495950059 +0800
@@ -134,8 +134,8 @@
 o = s:taboption("task", Value, "max_concurrent_downloads", translate("Max concurrent downloads"))
 o.placeholder = "5"
 
-o = s:taboption("task", Value, "max_connection_per_server", translate("Max connection per server"), "1-16")
-o.datetype = "range(1, 16)"
+o = s:taboption("task", Value, "max_connection_per_server", translate("Max connection per server"), "1-256")
+o.datetype = "range(1, 256)"
 o.placeholder = "1"
 
 o = s:taboption("task", Value, "min_split_size", translate("Min split size"), "1M-1024M")
